src/packagesystem.rs

```diff
@@ -67,12 +69,182 @@ where
     rpm_parse_metadata(&rpmout.stdout)
 }
 
-#[test]
-fn test_parse_rpmout() {
-    let testdata = "grub2-efi-x64-1:2.06-95.fc38.x86_64,1681321788 grub2-efi-x64-1:2.06-95.fc38.x86_64,1681321788 shim-x64-15.6-2.x86_64,1657222566 shim-x64-15.6-2.x86_64,1657222566 shim-x64-15.6-2.x86_64,1657222566";
-    let parsed = rpm_parse_metadata(testdata.as_bytes()).unwrap();
-    assert_eq!(
-        parsed.version,
-        "grub2-efi-x64-1:2.06-95.fc38.x86_64,shim-x64-15.6-2.x86_64"
-    );
+fn parse_evr_map(input: &str) -> BTreeMap<String, Evr> {
+    input
+        .split(',')
+        .map(|pkg| {
+            // assume it is like "grub2-1:2.12-28.fc42,shim-15.8-3"
+            if !pkg.ends_with(std::env::consts::ARCH) {
+                let (name, version_release) = pkg.split_once('-').unwrap_or((pkg, ""));
+                let evr = Evr::parse(version_release);
+                return (name.to_string(), evr);
+            }
+
+            // assume it is like "grub2-efi-x64-1:2.12-28.fc42.x86_64,shim-x64-15.8-3.x86_64"
+            let nevra = rpm::Nevra::parse(pkg);
```

Comment:
Per my earlier comment I think this is OK as is but I'd like to not add new hardcoded RPM dependencies. We can just make this a build-time configuration option in the future.

Reply:
Do you mean that we only import rpm if using `cargo build --features rpm`, else using `cargo build` without rpm, but we also use `cmp()` to compare `rpm::Evr` (see https://docs.rs/rpm/latest/rpm/struct.Evr.html#method.cmp), so we still need rpm, is this right?

Reply:
I think it's OK to have an rpm feature and use it.

But we could have a generic version that uses https://docs.rs/uapi-version/latest/uapi_version/ to compare versions and require that package systems map to a compatible version scheme (if it's not compatible).